x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/TimestampFormatFinder.java

```diff
@@ -288,6 +289,43 @@ public final class TimestampFormatFinder {
         )
     );
 
+    /**
+     * The first match in this list will be chosen, so it needs to be ordered
+     * such that more generic patterns come after more specific patterns.
+     */
+    static final List<CandidateTimestampFormat> ORDERED_CANDIDATE_FORMATS_ECS_V1;
+    static {
+        List<CandidateTimestampFormat> items = new ArrayList<>();
+        // CATALINA8_DATESTAMP %{MONTHDAY}-%{MONTH}-%{YEAR} %{HOUR}:%{MINUTE}:%{SECOND}
+        // 29-Aug-2009 12:03:33
+        items.add(
+            new CandidateTimestampFormat(
+                example -> Collections.singletonList("dd-MMM-yyyy hh:mm:ss"),
+                "\\b\\d{2}-[A-Z]\\S{2}-\\d{4} \\d{2}:\\d{2}:\\d{2}[:.,]\\d{3}",
+                "\\b%{MONTHDAY}-%{MONTH}-%{YEAR} %{HOUR}:%{MINUTE}:%{SECOND}\\b",
+                "CATALINA8_DATESTAMP",
+                "11     1111 11 11 11",
+                0,
+                13
+            )
+        );
+        // CATALINA7_DATESTAMP %{MONTH} %{MONTHDAY}, %{YEAR} %{HOUR}:%{MINUTE}:%{SECOND} (?:AM|PM)
+        items.add(
+            new CandidateTimestampFormat(
+                example -> Collections.singletonList("MMM dd, yyyy hh:mm:ss a"),
+                "\\b[A-Z]\\S{2} \\d{2}, \\d{2} \\d{4}:\\d{2}:\\d{2}[:.,]\\d{3}",
+                "\\b%{MONTH} %{MONTHDAY}, %{YEAR} %{HOUR}:%{MINUTE}:%{SECOND} (?:AM|PM)\\b",
+                "CATALINA7_DATESTAMP",
+                "    11  1111 11 11 11 111",
```

Comment:
```suggestion
                Arrays.asList("    11  1111 1 11 11", "    11  1111 11 11 11"),,
```

This one is basically the old `CATALINA_DATESTAMP`, but with the hardcoded `20` in the year replaced.